ilastik/workflows/carving/opCarving.py

```diff
@@ -556,37 +514,31 @@ def get_label_voxels(self):
 
         nonzeroSlicings = self.opLabelArray.NonzeroBlocks[:].wait()[0]
 
-        coors1 = [[], [], []]
-        coors2 = [[], [], []]
+        bg = [[], [], []]
+        fg = [[], [], []]
         for sl in nonzeroSlicings:
-            a = self.opLabelArray.Output[sl].wait()
-            w1 = numpy.where(a == 1)
-            w2 = numpy.where(a == 2)
-            w1 = [w1[i] + sl[i].start for i in range(1, 4)]
-            w2 = [w2[i] + sl[i].start for i in range(1, 4)]
+            label = self.opLabelArray.Output[sl].wait()
+            labels_fg = numpy.where(label == 1)
+            labels_bg = numpy.where(label == 2)
+            labels_fg = [labels_fg[i] + sl[i].start for i in range(1, 4)]
+            labels_bg = [labels_bg[i] + sl[i].start for i in range(1, 4)]
             for i in range(3):
-                coors1[i].append(w1[i])
-                coors2[i].append(w2[i])
+                bg[i].append(labels_fg[i])
+                fg[i].append(labels_bg[i])
```

Comment:
Apart from the actual code I'd suggest factoring out this method, or making it a staticmethod and write a test for it that only acts on a `OpLabelArray`. Side note: in future we should switch `opLabelArray` to a `OpCompressedUserLabelArray`...

Reply:
Nice catch... background=1 and foreground=2 simply because of the order in which they are introduced into the LabelNames slot (`["Background", "Object"]`). I've pushed fixing the flip for now, but will just extract those constants and then see if we rework the function itself a bit (it's not exactly critical for performance).

Reply:
oh yeah, nice catch @emilmelnikov!

Reply:
Updated in 298a35b4a32b5cb509b31b6c224ed8a87e8ba107 and 85686b5d3c339ee9a5bd5d529c5fa7db27ce1ccb